dockstore-client/src/main/java/io/dockstore/client/cli/nested/WesRequestData.java

```diff
@@ -68,8 +68,7 @@ public WesRequestData(String url, String awsAccessKey, String awsSecretKey, Stri
         this.awsSecretKey = awsSecretKey;
         this.region = region;
 
-        // The credentials that were passed in are null, so we are assuming they are within an authorized environment (such as an EC2 instance)
-        // This also assumes there is no scenario where just one of the keys is sufficient.
+        // AWS credentials are required to make a complete API call. If any of the parameters are null, throw an error for the user.
```

Comment:
Just to clarify, the no credentials case (instance profile, behind a firewall, etc.) will still work, right? I guess that's a different path; you just wouldn't specify auth at all?

Reply:
This blocks the no credential case, as an error will be thrown noting the user is missing either keys or a region if any inputs are null. I think the initial implementation of 'no credentials' was flawed.

If I'm understanding the documentation correctly, we should be able to setup a proper credential parsing chain using the AWS SDK. That chain will parse the EC2 instance metadata for IAM credentials (See [InstanceProfileCredentialsProvider](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/InstanceProfileCredentialsProvider.html)). 

AWS provides a large number of default credential providers, each parsing credentials from different locations and in different orders: [AWSCredentialsProvider](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentialsProvider.html), and we can chain together the ones we want.

That's a long way around of saying that for this ticket: https://ucsc-cgl.atlassian.net/browse/SEAB-3478, we should be able to add another credential provider that explicitly searches for EC2 instance credentials, and passes those to the `WesRequestData` object as normal. Because of this, if `WesRequestData` receives null AWS credentials as inputs, throwing an error seems reasonable.

Reply:
> This blocks the no credential case, as an error will be thrown noting the user is missing either keys or a region if any inputs are null.

The no-credentials case will still work for non-AWS WES, right? E.g., I'm running Cromwell locally as a WES Server and don't set up any authentication for it...

I think it's fine to hold off on the instance provider case, since you have the followup ticket.

Reply:
Ah, yes, no-credentials for non-AWS WES still works. `WesRequestData` accepts null bearer tokens, just not null AWS credentials.